### PR TITLE
[Ripple] Additional documentation for Ripple and its classes

### PR DESCRIPTION
**Pull request by yarneo**
As preparation for marking Ripple ready I made a pass and improved the documentation for Ripple and included missing documentation for its APIs.

### CODE REVIEW

**Comment by jverkoey** [comment]
Please confirm you've re-run `./scripts/generate_readme Ripple`. Some of the changes look like they may not have been made in the docs/ directory.

File: components/Ripple/README.md [hunk 1]
```diff
@@ -117,11 +139,11 @@ MDCRippleTouchController *rippleTouchController = [[MDCRippleTouchController all
 ```
 <!--</div>-->
 
-
 The `MDCRippleTouchControllerDelegate` gives you some control over aspects of the
-ripple/touch relationship, such as to temporarily disable ripple
-touches, the following code uses the delegate's
-`rippleTouchController:shouldProcessRippleTouchesAtTouchLocation:` method:
+ripple/touch relationship and its placement in the view hierarchy.
+In the below example we are using the delegate to declare we want to only process ripple
```
Comment (jverkoey) [inline components/Ripple/README.md]:
```suggestion
In the example below we are using the delegate to declare that we only want to process ripple 
```

File: components/Ripple/README.md [hunk 2]
```diff
@@ -117,11 +139,11 @@ MDCRippleTouchController *rippleTouchController = [[MDCRippleTouchController all
...
 <!--</div>-->
 
-
 The `MDCRippleTouchControllerDelegate` gives you some control over aspects of the
-ripple/touch relationship, such as to temporarily disable ripple
-touches, the following code uses the delegate's
-`rippleTouchController:shouldProcessRippleTouchesAtTouchLocation:` method:
+ripple/touch relationship and its placement in the view hierarchy.
+In the below example we are using the delegate to declare we want to only process ripple 
+touches if the touch is in a certain location, and also insert the Ripple view at the very bottom of
```
Comment (jverkoey) [inline components/Ripple/README.md]:
```suggestion
touches if the touch is in a certain location. We also insert the Ripple view at the bottom of
```

File: components/Ripple/README.md [hunk 3]
```diff
@@ -117,11 +139,11 @@ MDCRippleTouchController *rippleTouchController = [[MDCRippleTouchController all
...
+the parent view's view hierarchy:
```
Comment (jverkoey) [inline components/Ripple/README.md]: Consider explaining why we need to insert the ripple view at the bottom of the parent view's hierarchy here. Good opportunity to educate.

File: components/Ripple/README.md [hunk 4]
```diff
@@ -184,28 +227,111 @@ own touch processing:
...
 <!--</div>-->
 
+### MDCStatefulRippleView
+
+You can also use MDCStatefulRippleView to display stateful ripples using your
+own touch processing. 
+To fully benefit from MDCStatefulRipple's capabilities in moving through states visually,
+the view that is adding the stateful ripple view will need to override 
+UIView's `touchesBegan`, `touchesMoved`, `touchesEnded` and `touchesCancelled`
+and call the stateful ripple views corresponding APIs before calling the super.
```
Comment (jverkoey) [inline components/Ripple/README.md]:
```suggestion
and call the stateful ripple view's corresponding APIs before calling the super.
```

File: components/Ripple/README.md [hunk 5]
```diff
@@ -184,28 +227,111 @@ own touch processing:
...
-[inkView beginRippleTouchUpAnimated:YES completion:nil];
+[rippleView beginRippleTouchUpAnimated:YES completion:nil];
 ```
 <!--</div>-->
 
+### MDCStatefulRippleView
+
+You can also use MDCStatefulRippleView to display stateful ripples using your
+own touch processing. 
+To fully benefit from MDCStatefulRipple's capabilities in moving through states visually,
```
Comment (jverkoey) [inline components/Ripple/README.md]:
```suggestion
To fully benefit from MDCStatefulRipple's ability to move between states visually,
```

File: components/Ripple/README.md [hunk 6]
```diff
@@ -184,28 +227,111 @@ own touch processing:
...
+the view that is adding the stateful ripple view will need to override
```
Comment (jverkoey) [inline components/Ripple/README.md]:
```suggestion
the view that is adding the stateful ripple view must override 
```

File: components/Ripple/README.md (hunk 4 above)
Comment (jverkoey) [inline components/Ripple/README.md]:
```suggestion
and call the stateful ripple views corresponding APIs before calling the `super` implementation.
```

**Review by jverkoey — APPROVED** [review]

**Comment by yarneo** [comment]
I have re-run the generate_readme script but had to do manual changes to it as it was assuming the component isn't in beta. Shortly I will open a PR graduating it and hopefully will not need any manual changes after the generation

**Pull request merged**